### PR TITLE
Fix autosave scenarios for Template Editor

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Fix autosave scenarios for Template Editor

Some refactoring

## Motivation and Context
Should improve failure rates for e2e tests

## How Has This Been Tested?
No functionality changes. Tests are passing.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
N/A

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Main thing is the common `waitForAutosave` function for preventing autosave failures.

@ezequielc please review. Thanks!

File: test/e2e/template-editor/cases/components/financial.js [hunk 1]
```diff
@@ -39,12 +36,6 @@ var FinancialComponentScenarios = function () {
         templateEditorPage.dismissFinancialDataLicenseMessage();
       });
 
-      it('should auto-save the Presentation after it has been created', function () {
-        helper.waitDisappear(templateEditorPage.getDirtyText());
-        helper.waitDisappear(templateEditorPage.getSavingText());
-        helper.wait(templateEditorPage.getSavedText(), 'Financial component auto-saved');
```
Comment (ezequielc) [inline test/e2e/template-editor/cases/components/financial.js]: @alex-deaconu There are a few more instances of waiting for saving in this scenario below. Would it be helpful to use `templateEditorPage.waitForAutosave()` on those cases? Thanks

File: test/e2e/template-editor/cases/components/slides.js [hunk 2]
```diff
@@ -76,29 +71,31 @@ var SlidesComponentScenarios = function () {
...
+        //wait for presentation to be auto-saved
+        helper.wait(templateEditorPage.getSavedText(), 'RSS component auto-saved');
       });
 
-      it('should change the Presentation name and auto-save', function () {
-        // Note: sendKeys does not trigger the "ng-change" event for the duration textbox
-        // have to send a fake autoupdate to trigger it
-        presentationsListPage.changePresentationName(presentationName);
+      it('should clear and update the duration text', function () {
+        // Note: Disconnect from Angular to prevent Autosave timeout from interrupting edits
```
Comment (ezequielc) [inline test/e2e/template-editor/cases/components/slides.js]: @alex-deaconu Interesting idea.

File: test/e2e/template-editor/pages/presentationListPage.js [hunk 3]
```diff
@@ -37,6 +37,10 @@ var PresentationListPage = function() {
 
     helper.wait(templateEditorPage.getTemplateEditorContainer(), 'Template Editor Container');
     browser.sleep(500);
+
+    helper.waitDisappear(templateEditorPage.getDirtyText());
+    helper.waitDisappear(templateEditorPage.getSavingText());
+    helper.wait(templateEditorPage.getSavedText(), 'Template auto-saved');
```
Comment (ezequielc) [inline test/e2e/template-editor/pages/presentationListPage.js]: @alex-deaconu Can we replace this with `templateEditorPage.waitForAutosave()`? Thanks

File: test/e2e/template-editor/cases/components/financial.js (hunk 1 above)
Comment (alex-deaconu) [reply test/e2e/template-editor/cases/components/financial.js]: @ezequielc yes, most likely. The main reason I did not update with `waitForAutosave` is because these patterns have `waitDisappear > waitDisappear > wait` while the `waitForAutosave` pattern is `waitDisappear > wait > wait`.

Main difference is that the first pattern does not expect an Autosave, but in the case there is one in progress it handles it.

However, with my last commit I'm fixing that expectation. I did not update the rest of the `waitDisappear > waitDisappear > wait` patterns because technically there's no Autosave expected there and should not be needed in the first place.

Long story short we can pretty much add `waitForAutosave` anywhere right now, but I tried to restrict it to actual places where we want to Autosave to validate the saved results.

Let me know if you'd like to discuss this further and how you advise we proceed.

Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc updated as discussed. Thanks!

**Comment by ezequielc** [comment]
@alex-deaconu Looks good. Thanks 👍
